core/include/prometheus/counter.h

```diff
@@ -45,7 +63,37 @@ class Counter {
   ClientMetric Collect() const;
 
  private:
-  Gauge gauge_{0.0};
+  int ThreadId() {
+    thread_local int id{-1};
+
+    if (id == -1) {
+      id = AssignThreadId();
+    }
+    return id;
+  }
+
+  int AssignThreadId() {
+    const int id{count_.fetch_add(1)};
+
+    if (id >= per_thread_counter_.size()) {
+      std::terminate();
```

Comment:
You may initialize zero element in `per_thread_counter_` and use it if threads count goes beyond of array. But this does not work with load/store scheme. Why are not you using fetch_add with relaxed policy?